client/src/components/pages/TeacherProfile/index.js

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from 'react';
+import './index.css';
+import TeacherImage from '../../../assets/teacher.jpg';
+import fakeData from './fakeData';
+import teacherTape from '../utils/teacherTap';
+import renderButton from '../utils/renderButton';
+
+class TeacherProfile extends Component {
+  state = {
+    data: {},
+    activeContent: `about`,
+  };
+
+  componentDidMount() {
+    this.setState({ data: fakeData.teacher1 });
+    // we will make a request to get data by using axios, above line is a fake data
+  }
+
+  showContent = tapName => {
+    this.setState({ activeContent: tapName });
+  };
+
+  render() {
+    const { activeContent } = this.state;
+    const { data } = this.state;
+
+    return (
+      <div className="teach-profile">
+        <div>
+          <img className="img-teacher" src={TeacherImage} alt="teacher img" />
+        </div>
+        <h3 className="teacher-name">{data.name}</h3>
+        <p className="teacher-subject">{data.subject} Teacher</p>
+        <div className="teach-profile__list">
+          {renderButton(activeContent, 'About', this.showContent)}
+          {renderButton(activeContent, 'Contact', this.showContent)}
+          {renderButton(activeContent, 'Interests', this.showContent)}
```

Comment:
going the right direction but still some issues. Notice here that the strings start with capital letters, whereas in `teacherTape` you're using lower case letters in the comparison: `if (num === 'contact') {`
They will thus never match. This is why it's good to always define constants for hardcoded values. You can do that at the top of the file:
```
const var ABOUT = "About";
etc..
```
and use it here and then in the comparison.

Reply:
Good, I'll change these things.
but before I used text.toLowerCase(). what's better use function to get the same string, Or define it as constants

Reply:
Constant is better. If in the future the value changes, you just update the constant, otherwise you'd have to search through the code and change the value everywhere.